tests/commands/hook_impl_test.py

```diff
@@ -216,7 +216,7 @@ def test_run_ns_pre_push_new_branch(push_example):
         ns = hook_impl._run_ns('pre-push', False, args, stdin)
 
     assert ns is not None
-    assert ns.from_ref == src_head
```

Comment:
Why did this change?

Reply:
The PR as currently written ensures that creating a new branch will always set `PRE_COMMIT_FROM_REF=00000...` and `PRE_COMMIT_TO_REF=1234deadbeef...`. The thing that pre-commit upstream is _trying_ to do is determine a "primary" branch divergence point, but Git has no real concept of that, so instead it uses _all the remote's existing branches_ as a proxy. Once a branch is created, it's easy to say "we need to run checks on files that modified from here to here", but during branch creation, we literally have no reference point. I originally tried kludging to preserve behavior, but the resulting algorithm was really inconsistent and hard to reason about.

I could try to update these tests to _update_ a remote ref, if you like.

Reply:
Ah, interesting. So we're changing pre-commit's behavior here because it's it uses potentially not-cogent heuristics? IIUC we don't have any pre-push hooks other than the one you wrote recently, so this should be safe to do. Can we call this out/document these behavior changes somewhere for posterity?

Reply:
> So we're changing pre-commit's behavior here because it's it uses potentially not-cogent heuristics?

Yes.

> Can we call this out/document these behavior changes somewhere for posterity?

Added a long-ish comment to `_pre_push_ns`.